Created Explore Page Card

laurasofi1024 opened the pull request:
Established Bare-Bones

laurasofi1024 commented:
**Opened a new pull request to merge to frontend Branch instead of main.**

Pending comment from @erin2722 from the previous PR:

> Looks good so far! Here is the advice about icons:
> 
> The team is going to use font awesome to get icons so that it is easier for us to get icons and change their colors. At https://fontawesome.com/icons/edit?style=solid you can search for the icons. I think the 3 for the box will be user, check-circle, and edit. I pushed a change to frontend showing how to use them!
> 
> Let me know if you have any questions!

laurasofi1024 commented:
Current Mobile View:

![localhost_3000_explore(iPhone X) (1)](https://user-images.githubusercontent.com/61126997/97658140-fa54f400-1a41-11eb-8832-670f89555dde.jpg)

Current Desktop View:
![Screen Shot 2020-10-29 at 11 52 23 PM](https://user-images.githubusercontent.com/61126997/97658177-0d67c400-1a42-11eb-82d4-cbfbdc123cf1.png)

erin2722 commented on src/components/explorebox/index.js:
```diff
@@ -0,0 +1,190 @@
+                    <hr/>
+                    <Row>
+                        <Item>
+                            <div><FontAwesomeIcon icon={faUser}></FontAwesomeIcon></div>
```
super nitpicky, but if you wanted to cut down on divs you could remove them here and change the div style inside Item to svg and switch padding to margin. The more semantic your HTML the better! Definitely don't need to do this though

erin2722 commented on src/components/explorebox/index.js:
```diff
@@ -0,0 +1,190 @@
+                            <h2>{name}</h2>
+                            <p>{description}</p>
+                        </div>
+                        <Icon>{imageURL}</Icon>
```
instead of the image url prop being a tag, could we have it just be the url and then here do:
```HTML
<img alt='club picture' src={imgURL} />
```

erin2722 replied:
just so that all the html is in the same place I think it makes it a bit more readable

erin2722 commented on src/components/explorebox/index.js:
```diff
@@ -0,0 +1,190 @@
+                        </div>
+                        <Icon>{imageURL}</Icon>
+                    </Row>
+                    <Tags>{tags}</Tags>
```
I think that the tags that etesam made should be in the frontend branch, do you think we could integrate them here?

erin2722 commented on src/components/explorebox/index.js:
```diff
@@ -0,0 +1,190 @@
+`;
+
+
+export default class ExploreBox extends React.Component{
```
just so that we have consistent conventions throughout the codebase, could this be a hook? For ex:
```Javascript
export const ExploreBox = (name, description, clubSize, tags, imageURL, acceptingMemebers, applicationRequired, cardLink) => {
...
};

erin2722 replied:
typing this out also made me realized that we might just want to pass the entire club object in instead of each individual prop, but that is a small fix for later

erin2722 reviewed (changes requested):
This looks awesome!!! 🎉 🎉 

I'd say once you integrate the tags from frontend, change this to a React Hook, and fix the imgURL prop, we are good to merge this in!

I literally said to myself "wow, this looks incredible," so good job! 😄 👍

laurasofi1024 commented:
Final Fixes: integrated the tags from frontend, changed to React Hook, and fixed the imgURL prop; also made small changes to margins of tags from frontend

laurasofi1024 commented:
Current Desktop View:
![Screen Shot 2020-11-03 at 5 52 59 PM](https://user-images.githubusercontent.com/61126997/98050248-436cc580-1dff-11eb-9c84-7d90830fb39b.png)

Current Mobile View (iPhoneX):
![localhost_3001_explore(iPhone X)](https://user-images.githubusercontent.com/61126997/98050219-33ed7c80-1dff-11eb-8fbc-0a522f6c86c9.png)

erin2722 commented on client/src/containers/Explore.js:
```diff
@@ -1,9 +1,103 @@
+                        name = "Columbia University Jazz Ensemble" 
+                        description = "A small advanced jazz band covering 1950's hard bop to more adventurous contemporary Avant Garde styles."
+                        imageURL = 'https://testbucket102920.s3.amazonaws.com/Ellipse+5+(2).png'
+                        tags = {[<SearchTag text='Music'/>, <SearchTag text='Performing Arts'/>, <SearchTag text='Community Service'/>, <SearchTag text='Global Affairs'/>, <SearchTag text='Pre-professional'/>]}
```
instead of inputting the tags as an array of tags, could you just pass in the names as an array and then do 
``` javascript
tags.map(tag => <SearchTag text={tag} />);
```
to display all of the tags? just so that it will be easier to map different tag array sizes

erin2722 replied:
oh! don't delete .github

erin2722 reviewed (changes requested):
This looks awesome!

Could you change the tag mapping and delete the extra files outside of the client folder (clubs-2020, .github, src/, package-lock) when you repush?

After that we will get this merged in!

erin2722 reviewed (approved):
🎉
